realm/realm-library/src/androidTest/java/io/realm/RealmIntegerTests.java

```diff
@@ -233,54 +363,125 @@ public void copyFromIsNotLive() {
     }
 
     /**
-     * Assure that an attempt to change the value of a managed RealmInteger, outside a transaction, fails.
+     * Assure that an attempt to assign a RealmInteger field, outside a transaction, fails.
      */
-    @Ignore("not yet implemented")
     @Test
-    public void updateOutsideTransactionThrows() {
+    public void assignOutsideTransactionThrows() {
         realm.beginTransaction();
-        realm.createObject(AllTypes.class).setColumnRealmInteger(new RealmInteger(5));
+        Counters managedCounters = realm.createObject(Counters.class);
+        managedCounters.setColumnRealmInteger(RealmInteger.valueOf(5));
         realm.commitTransaction();
 
-        RealmInteger managedRI = realm.where(AllTypes.class).findFirst().getColumnRealmInteger();
         try {
-            managedRI.set(1);
-            fail("Setting a managed RealmInteger outside a transaction should fail");
+            managedCounters.setColumnRealmInteger(RealmInteger.valueOf(7));
+            fail("Assigning a managed RealmInteger outside a transaction should fail");
         } catch (IllegalStateException e) {
             assertTrue(e.getMessage().contains("must be in a transaction"));
         }
+    }
+
+    /**
+     * Semantics definition from #4266.
```

Comment:
Complete link is also good, as long as it easy to reach the discussion. I think adding them in {@link x} tags makes it possible to jump directly to it?